azavea/temperate
# Feature/jf/risks as assessed

## Overview

Adds check methods to tell if a risk is considered fully assessed, and updates where appropriate. Allows nulls on certain fields on `OrganizationalRisk` to make their values truly user-selected.

### Demo
![397](https://user-images.githubusercontent.com/10568752/34896771-173cb264-f7b9-11e7-8346-fcad028a6784.gif)


<img width="637" alt="screen shot 2018-01-12 at 4 58 29 pm" src="https://user-images.githubusercontent.com/10568752/34896982-dbbad404-f7b9-11e7-86f5-1b26009e5848.png">


### Notes

I initially tried adding a method to the front-end `Risk` class. I learned that typing doesn't mean that properties on the type class are applied, so this was a dead end when handling risks returned from the database, which is everywhere.

I considered leaving the method for use when applicable (on risks created directly from the`Risk` class) but thought that would be more confusing than helpful.

I didn't touch the `OrganizationalRisk` Django model for this because to my understanding, the front end is the one that cares about such information.

## Testing Instructions

`./scripts/manage migrate`
Create a pretty barebones risk with null values. See that
- [ ] only if the assessed field(s) are filled out should "Continue" be unlocked. IDK if there should be any visual cue that those are *the* fields.
- [ ] Fields should start null that before started "Unsure".
- [ ] There should be no errors saving null fields.
- [ ] your created risk in the risk list should have the green assess button

Closes #397, #405
